### PR TITLE
add t_inf to sc doses in new_regimen

### PR DESCRIPTION
**Pull request by JordanBrooks33**
Added t_inf argument to new_regimen() and adapted tests.

### CODE REVIEW

**Comment by JordanBrooks33** [comment]
@jasmineirx Thank you! That makes a lot of sense. I have updated the logic to assign the correct t_inf default per type. I also included im alongside sc as a default of 1/60, given the similarity of these routes. I also updated the tests you had added to be slightly broader, but still testing the same scenarios.

File: R/new_regimen.R [hunk 1]
```diff
@@ -69,9 +69,29 @@ new_regimen <- function(
...
+    }
+    if(any(type == "bolus") && (is.null(t_inf) || length(t_inf) == 0)) {
+      reg$t_inf[reg$type=="bolus"] = 0
+    } else if (any(is.na(t_inf))) {
+      reg$t_inf[(reg$type=="bolus" & is.na(reg$t_inf))] <- 0
+    }
+    if(any(type == "oral") && (is.null(t_inf) || length(t_inf) == 0)) {
+      reg$t_inf[reg$type=="oral"] = 0
+    } else if (any(is.na(t_inf))) {
+      reg$t_inf[(reg$type=="oral" & is.na(reg$t_inf))] <- 0
```
Comment (jasmineirx) [inline R/new_regimen.R:94]: the blocks above are pretty repetitive.

what about something like

```r

if (is.null(t_inf) || length(t_inf) == 0) {
  reg$t_inf <- rep(NA_real_, length(reg$type))
}
if (any(is.na(reg$t_inf))) {
  # impute using standard values for each dose type
  impute_t_inf <- c(infusion = 1, sc = 1/60, im = 1/60, oral = 0, bolus = 0)
  idx <- which(is.na(reg$t_inf))
  reg$t_inf[idx] <- impute_t_inf[reg$type[idx]]
  # set unrecognized types to 1
  reg$t_inf[is.na(reg$t_inf)] <- 1
}
```

**Review by jasmineirx — COMMENTED** [review]
I have a suggestion for reducing the repetitiveness of the code while balancing readability

File: R/new_regimen.R (hunk 1 above)
Comment (JordanBrooks33) [reply R/new_regimen.R:94]: very nice.

**Review by jasmineirx — APPROVED** [review]
lgtm!

**Review by roninsightrx — APPROVED** [review]
lgtm

**Pull request merged**